### PR TITLE
Performance Improvement

### PR DESCRIPTION
**Pull request by DilsherB**
## Screen Shots.
data inserted into visits
inserted data into owners
animal_id = 4
vet_id = 2
email
indexes created
animal_id = 4 after index
vet_id = 2 after index
email after index

## In this PR we did the following:

1. Added a column in the owners table.
2. Inserted more data in visits and owners table.
3. three new queries were added.
4. created non-clustered indexes to boost the performance of newly created queries.
5. Check the performance by `explain analyze` tool.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @DilsherB 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show commit from your partner and also your PR is not descriptive Kindly Write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: schema.sql [hunk 1]
```diff
@@ -44,4 +44,6 @@ CREATE TABLE visits(
   animal_id INT REFERENCES animals(id),
   vet_id INT REFERENCES vets(id),
   date_of_visit date
-);
\ No newline at end of file
+);
+
+ALTER TABLE owners ADD COLUMN email varchar(120);
```
Comment (Gambit142) [inline schema.sql]: - [x] Kindly add an index to the following in this file:
- In the `visits` table, the `animal_id` column
- In the `visits` table, the `vet_id` column
- In the `owners` table, the `email` column

Adding indexes to columns sometimes optimises database queries and it is an efficient way to retrieve a relatively small number of rows from a table. Kindly implement this change. 👍 💻

File: data.sql [hunk 2]
```diff
@@ -104,4 +104,21 @@ VALUES
...
+    owners (full_name, email)
+    select 'Owner ' ||
+    generate_series(1,2500000), 'owner_' ||
+    generate_series(1,2500000) || '@mail.com';
+
+INSERT INTO
+    visits (animal_id, vet_id, date_of_visit)
+    SELECT * FROM (SELECT id FROM animals) animal_ids,
+    (SELECT id FROM vets) vets_ids,
+    generate_series('1980-01-01'::timestamp, '2021-01-01', '4 hours') visit_timestamp;
```
Comment (Gambit142) [inline data.sql]: - [x] Great work inserting data the way it was recommended. However, I noticed in your screenshots that the execution time before improving performance is significantly lower than **_1 second (1000ms)_**. I would advise you to run this query two or three times, make the `explain analyze` statement and take screenshots of the execution time that are more than _**1 second (1000ms)**_. 👍 😎 

#### You can check points _**3 and 4**_ of this [project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_performance_audit.md) 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @DilsherB @lenkon 👋, 

Good job so far!

## Highlights
- Populated database with data ✔️ 
- Created index on required columns ✔️ 
- Use of the `explain analyze` command ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: schema.sql (hunk 1 above)
Comment (DilsherB) [reply schema.sql]: @Gambit142, changes done

File: data.sql (hunk 2 above)
Comment (DilsherB) [reply data.sql]: @Gambit142, changes done

**Review by Davidosky007 — APPROVED** [review]
Hi @DilsherB @lenkon 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Database performance improved 👍
* Well written PR 👍
* Good looking readme 👍
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**